app/validators/base_path_state_validator.rb

```diff
@@ -0,0 +1,53 @@
+class BasePathStateValidator < ActiveModel::Validator
+  def validate(record)
+    return unless record.content_item
+
+    content_item = record.content_item
+    state = content_item_state(record)
+    base_path = content_item_base_path(record)
+
+    return unless state && base_path
+
+    check_draft_conflict(record, content_item, base_path) if state == "draft"
+    live = content_item_live?(content_item, state)
+    check_live_conflict(record, content_item, base_path) if live
+  end
+
+private
+
+  def content_item_state(record)
+    return record.name if record.is_a?(State)
+    State.where(content_item: record.content_item).pluck(:name).first
+  end
+
+  def content_item_base_path(record)
+    return record.base_path if record.is_a?(Location)
+    Location.where(content_item: record.content_item).pluck(:base_path).first
+  end
+
+  def content_item_live?(content_item, state)
+    return true if state == "published"
+    state == "unpublished" && Unpublishing.is_substitute?(content_item)
```

Comment:
I think this should be `state == "unpublished" && !Unpublishing.is_substitute?(content_item)` (notice the `!`), as you want to say that all types of unpublishing are "live", except "substitute".